src/components/item/header/MobileItemHeaderActions.tsx

```diff
@@ -0,0 +1,181 @@
+import DownloadButton from '@/components/main/DownloadButton';
+import { ITEM_ID_PARAMS } from '@/config/paths';
+
+import { ITEM_TYPES_WITH_CAPTIONS } from '../../../config/constants';
+import { useBuilderTranslation } from '../../../config/i18n';
+import { hooks } from '../../../config/queryClient';
+import {
+  ITEM_CHATBOX_BUTTON_ID,
+  MOBILE_MORE_ACTIONS_BUTTON_ID,
+} from '../../../config/selectors';
+import { ItemActionTabs } from '../../../enums';
+import { BUILDER } from '../../../langs/constants';
+import {
+  getHighestPermissionForMemberFromMemberships,
+  isItemUpdateAllowedForUser,
+} from '../../../utils/membership';
+import ItemMetadataButton from '../../common/ItemMetadataButton';
+import { useCurrentUserContext } from '../../context/CurrentUserContext';
+import { useLayoutContext } from '../../context/LayoutContext';
+import MobileItemMenu from '../../main/MobileItemMenu';
+import ItemSettingsButton from '../settings/ItemSettingsButton';
+import ModeButton from './ModeButton';
+
+const { useItemMemberships } = hooks;
+const { useItem } = hooks;
+
+export const MobileItemHeaderActions = (): JSX.Element => {
+  const { t: translateBuilder } = useBuilderTranslation();
+  const itemId = useShortenURLParams(ITEM_ID_PARAMS);
+
+  const { data: item } = useItem(itemId);
+
+  const {
+    editingItemId,
+    openedActionTabId,
+    isChatboxMenuOpen,
+    setIsChatboxMenuOpen,
+  } = useLayoutContext();
+
+  const { data: member } = useCurrentUserContext();
+  const [isItemActionsDrawerOpen, setIsItemActionsDrawerOpen] = useState(false);
+  const theme = useTheme();
+
+  const { data: memberships } = useItemMemberships(item?.id);
+  const canEdit = isItemUpdateAllowedForUser({
+    memberships,
+    memberId: member?.id,
+  });
+  const canAdmin = member?.id
+    ? getHighestPermissionForMemberFromMemberships({
+        memberships,
+        memberId: member?.id,
+      })?.permission === PermissionLevel.Admin
+    : false;
+
+  const onClickChatbox = () => {
+    setIsChatboxMenuOpen(!isChatboxMenuOpen);
+  };
+
+  const toggleActionsDrawer = () => {
+    setIsItemActionsDrawerOpen(!isItemActionsDrawerOpen);
+  };
+  const closeDrawer = () => setIsItemActionsDrawerOpen(false);
+  const renderItemActions = () => {
+    // if id is defined, we are looking at an item
+    if (item && item?.id) {
+      // show edition only for allowed types
+      const showEditButton =
+        !editingItemId &&
+        ITEM_TYPES_WITH_CAPTIONS.includes(item.type) &&
+        canEdit;
+
+      const activeActions = (
+        <>
+          {showEditButton && (
+            <EditButton
+              item={item}
+              onClick={closeDrawer}
+              type={ActionButton.MENU_ITEM}
+            />
+          )}
+          <Box width="100%">
+            <MobileItemMenu
+              item={item}
+              canEdit={showEditButton}
+              closeDrawer={closeDrawer}
+              canAdmin={canAdmin}
+            />
```

Comment:
Why are actions split between here and the `MobileItemMenu` ?? I do not think it makes sens to split into intermediary components in this way. It makes it very irregular and hard to navigate. 
Regularity and elegant components simplify the reading and understanding code process. 

As this PR has been open for some time already, I think it would be good for you to make a self review of the code, so you can examine your changes. Try to refine and sculpt your changes so they make sense with the current specifications. 

It is inherent to the process of writing code that the scope changes, and the way you have architected your code on the first try, can be re-modeled when you look back at the full-picture. 

I think this refining step is crucial to remove deadweight and make your components efficient and easy to read. I think doing an auto-review is always a good thing.